integration-tests/src/test/java/org/zowe/apiml/integration/graphql/BookControllerTest.java

```diff
@@ -72,13 +79,396 @@ public void whenGetAllBooksWithWrongSchema_thenReturnError() throws SSLException
+
+        // add book which will be later deleted
+        Book bookToDelete = setUpBook();
+        String addBookDocument = String.format("""
+         mutation {
+            addBook(name: "%s", pageCount: %d, authorId: "%s") {
+                bookId
+                name
+                pageCount
+            }
+        }
+        """, bookToDelete.name, bookToDelete.pageCount, bookToDelete.authorId );
+
+        HttpGraphQlTester.Response addBookResponse = tester.document(addBookDocument)
+            .execute();
+        String addedBookId = addBookResponse.path("addBook.bookId").entity(String.class).get();
+
+        //delete book
+        String deleteBookDocument = String.format("""
+         mutation {
+            deleteBook(bookId: "%s") {
+                bookId
+                name
+                pageCount
+            }
+        }
+        """, addedBookId);
+        tester.document(deleteBookDocument)
+            .execute()
+            .path("deleteBook")
+            .entity(Book.class)
+            .satisfies(book -> {
+                assertEquals(addedBookId, book.bookId);
+                assertEquals(bookToDelete.name, book.name);
+                assertEquals(bookToDelete.pageCount, book.pageCount);
+            });
+
+        //verify that book was deleted
+        String getBookByIdDocument = String.format("""
+        query {
+            getBookById(bookId: "%s") {
+                bookId
+                name
+                pageCount
+            }
+        }
+        """, addedBookId);
+
+        tester.document(getBookByIdDocument)
+            .execute()
+            .path("getBookById")
+            .valueIsNull();
+    }
+
+    @Test
+    public void whenDeleteUnknownBook_thenReturnException() throws SSLException {
+        SslContext sslContext = SslContextBuilder
+            .forClient()
+            .trustManager(InsecureTrustManagerFactory.INSTANCE)
+            .build();
+        HttpClient httpClient = HttpClient.create().secure(t -> t.sslContext(sslContext));
+
+        WebTestClient client =
+            WebTestClient.bindToServer().clientConnector(new ReactorClientHttpConnector(httpClient))
+                .baseUrl("https://localhost:10010/discoverableclient/api/v1/graphql")
+                .build();
+        HttpGraphQlTester tester = HttpGraphQlTester.create(client);
+
+        String unknownId = "unknown-id";
+        String deleteBookDocument = String.format("""
+         mutation {
+            deleteBook(bookId: "%s") {
+                bookId
+                name
+                pageCount
+            }
+        }
+        """, unknownId);
+
+        AssertionError thrown = assertThrows(AssertionError.class, () -> {
+            tester.document(deleteBookDocument)
+                .execute()
+                .path("deleteBook")
+                .entity(Book.class)
+                .get();
+        });
+        assertNotNull(thrown);
+    }
+
+    static private class Book {
```

Comment:
You can consider using Lombok annotations (`@Data`, `AllArgsConstructor`) to reduce the code.